File: Exercises/MPI/Examples/ex2.f
      program shifter
      implicit none

      include 'mpif.h'

      integer my_pe_num, errcode, numbertoreceive, numbertosend
      integer status(MPI_STATUS_SIZE)

      call MPI_INIT(errcode)

      call MPI_COMM_RANK(MPI_COMM_WORLD, my_pe_num, errcode)

      numbertosend = 42

      if (my_PE_num.EQ.0) then

         call  MPI_Recv( numbertoreceive, 1, MPI_INTEGER, 1,
     +                   10, MPI_COMM_WORLD, status, errcode)

         print *, 'Number received is: ',numbertoreceive

      endif


      if (my_PE_num.EQ.1) then

         call MPI_Send( numbertosend, 1, MPI_INTEGER, 0, 10,
     +                  MPI_COMM_WORLD, errcode)

      endif

      call MPI_FINALIZE(errcode)

      end

File: Exercises/MPI/Examples/ex2.c
#include <stdio.h>
#include "mpi.h"

main(int argc, char** argv){

  int my_PE_num, numbertoreceive, numbertosend=42;
  MPI_Status status;

  MPI_Init(&argc, &argv);
  MPI_Comm_rank(MPI_COMM_WORLD, &my_PE_num);

  if (my_PE_num==0){
    MPI_Recv( &numbertoreceive, 1, MPI_INT, MPI_ANY_SOURCE, MPI_ANY_TAG,
                MPI_COMM_WORLD, &status);
    printf("Number received is: %d\n", numbertoreceive);}

  else MPI_Send( &numbertosend, 1, MPI_INT,
			     0, 10, MPI_COMM_WORLD);
  MPI_Finalize();

}
